core/domain/user_services_test.py

```diff
@@ -642,6 +819,88 @@ def test_get_user_ids_by_role(self):
+        user_services.update_user_bio(user_id, user_bio)
+        user_settings = user_services.get_user_settings_from_email(user_email)
+
+        self.assertEqual(user_bio, user_settings.user_bio)
+
+    def test_update_preferred_language_codes(self):
+        auth_id = 'someUser'
+        user_email = 'user@example.com'
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+        user_settings_model = user_models.UserSettingsModel.get_by_id(user_id)
+        user_services.update_preferred_language_codes(
+            user_id, user_settings_model.preferred_language_codes)
+        user_settings = user_services.get_user_settings_from_email(user_email)
+
+        self.assertEqual(
+            user_settings_model.preferred_language_codes,
+            user_settings.preferred_language_codes
+        )
+
+    def test_update_preferred_site_language_code(self):
+        auth_id = 'someUser'
+        user_email = 'user@example.com'
+        preferred_site_language_code = 'en'
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+        user_services.update_preferred_site_language_code(
+            user_id, preferred_site_language_code)
+        user_settings = user_services.get_user_settings_from_email(user_email)
+
+        self.assertEqual(
+            preferred_site_language_code,
+            user_settings.preferred_site_language_code
+        )
+
+    def test_update_preferred_audio_language_code(self):
+        auth_id = 'someUser'
+        user_email = 'user@example.com'
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+        user_settings_model = user_models.UserSettingsModel.get_by_id(user_id)
+        user_services.update_preferred_audio_language_code(
+            user_id, user_settings_model.preferred_audio_language_code)
+        user_settings = user_services.get_user_settings_from_email(user_email)
+
+        self.assertEqual(
+            user_settings_model.preferred_audio_language_code,
+            user_settings.preferred_audio_language_code
+        )
+
+    def test_remove_user_role(self):
+        auth_id = 'someUser'
+        user_email = 'user@example.com'
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+        user_settings_model = user_models.UserSettingsModel.get_by_id(user_id)
+        user_services.add_user_role(user_id, feconf.ROLE_ID_BLOG_POST_EDITOR)
+        user_settings = user_services.get_user_settings_from_email(user_email)
+
+        user_services.remove_user_role(user_id, feconf.ROLE_ID_BLOG_POST_EDITOR)
+
+        self.assertEqual(
+            user_settings_model.roles,
+            user_settings.roles
+        )
+
+    def test_remove_user_role_for_default_role_raises_error(self):
+        auth_id = 'someUser'
+        user_email = 'user@example.com'
+        error_msg = 'Removing a default role is not allowed.'
```

Comment:
One general comment: in places where you use a constant once (and there's no complex logic), consider just inlining it, so that the reader does not need to carry around so much state in their head.

This applies in a few other places.

Reply:
Done